chart/templates/networkpolicy.yaml

```diff
@@ -78,21 +78,25 @@ spec:
     matchLabels:
       release: {{ .Release.Name }}
   egress:
-    # Allows outgoing connections to all pods with
-    # port 443, 8443 or 6443. This is needed for host Kubernetes
-    # access
+    # Allows outgoing connections to all necessary pods with
+    # port 443, 8443 or 6443 or system dns. This is needed for
+    # host Kubernetes access.
     - ports:
         - port: 443
         - port: 8443
         - port: 6443
-    # Allows outgoing connections to all vcluster workloads
-    # or kube system dns server
+        - port: 53
+          protocol: UDP
+        - port: 53
+          protocol: TCP
```

Comment:
Since we already have the rule below that selects DNS in the kube-system namespace, why is this still required?

Reply:
The previous NetworkPolicy `vc-work-{{ .Release.Name }}` will work on **workload node** based on its podSelector:

```
  podSelector:
    matchLabels:
      vcluster.loft.sh/managed-by: {{ .Release.Name }}
```

And, the control-plane of a vcluster still need to communicate **host's kube-dns**. That is why we want add this rule. And it did crash without this rule.